ergo-lib/src/wallet/multi_sig.rs

```diff
@@ -0,0 +1,914 @@
+              "assets": [],
+              "additionalRegisters": {},
+              "creationHeight": 0,
+              "transactionId": "6e32d1710816be34fd9710148b73f017bf8e71115cd2d3cf5758f80c2e3010ca",
+              "index": 2
+            }
+          ]
+        }"#;
+        let bytes_m = Base16DecodedBytes::try_from("100208cd03c847c306a2f9a8087b4ae63261cc5acea9034000ba8d033b0fb033247e8aade908cd02f4b05f44eb9703db7fcf9c94b89566787a7188c7e48964821d485d9ef2f9e4c4ea0273007301").unwrap();
+        let tree_m: ErgoTree = ErgoTree::sigma_parse_bytes(&bytes_m.0).unwrap();
+
+        let contx = Rc::new(force_any_val::<Context>());
+        let exp = tree_m.proposition().unwrap();
+        let reduction_result = reduce_to_crypto(&exp, &Env::empty(), contx).unwrap();
+        let sigma_tree = reduction_result.sigma_prop;
+        let stx: Transaction = serde_json::from_str(signed_tx).unwrap();
+        let test: ProofBytes = stx.inputs.first().clone().spending_proof.proof;
+        let proof: Vec<u8> = Vec::from(test);
+        let mut real_proposition: Vec<SigmaBoolean> = Vec::new();
+        let mut simulated_proposition: Vec<SigmaBoolean> = Vec::new();
+        let mut bag: HintsBag = bag_for_multi_sig(
+            sigma_tree.clone(),
+            real_proposition.as_slice(),
+            simulated_proposition.as_slice(),
+            proof.as_slice(),
+        )
+        .unwrap();
+        assert!(bag.hints.is_empty(), "{}", "{}");
+
+        let address_encoder = AddressEncoder::new(NetworkPrefix::Mainnet);
+        let first_address: Address = address_encoder
+            .parse_address_from_str("9hz1anoLGZGf88hjjrQ3y3oSpSE2Kkk15CcfFqCNYXDPWKV6F4i")
+            .unwrap();
+        let second_address = address_encoder
+            .parse_address_from_str("9gNpmNsivNnAKb7EtGVLGF24wRUJWnEqycCnWCeYxwSFZxkKyTZ")
+            .unwrap();
+
+        if let Address::P2Pk(pk) = first_address {
+            {
+                real_proposition.push(SigmaBoolean::ProofOfKnowledge(
+                    SigmaProofOfKnowledgeTree::ProveDlog(pk),
+                ));
+            }
+        }
+        if let Address::P2Pk(pk) = second_address {
+            {
+                simulated_proposition.push(SigmaBoolean::ProofOfKnowledge(
+                    SigmaProofOfKnowledgeTree::ProveDlog(pk),
+                ));
+            }
+        }
+        bag = bag_for_multi_sig(
+            sigma_tree,
+            real_proposition.as_slice(),
+            simulated_proposition.as_slice(),
+            proof.as_slice(),
+        )
+        .unwrap();
+        assert!(!bag.hints.is_empty(), "{}", "{}");
+    }
+
+    #[test]
+    fn generating_commitment_two_signer() {
+        let secret1 = DlogProverInput::random();
+        let secret2 = DlogProverInput::random();
```

Comment:
`secret2` should be `DhTupleProverInput` according to 
https://github.com/ScorexFoundation/sigmastate-interpreter/blob/795c176debfe105646a7deef3f01bd537b79c81d/sigmastate/src/test/scala/sigmastate/utxo/ProverSpecification.scala#L22

Reply:
i changed it in new commits